src/lib/components/tablev2/Search.js

```diff
@@ -0,0 +1,92 @@
+//@flow
+import React from 'react';
+import styled from 'styled-components';
+import { useTableContext } from './Tablev2.component';
+import SearchInput from '../searchinput/SearchInput.component';
+import { BasicText } from '../text/Text.component';
+import { spacing } from '../../style/theme.js';
+
+export type DisplayedName = {
+  plural: string,
+  singular: string,
+};
+
+export type SearchProps = {
+  onChange: (string) => void,
+  value?: string,
+  displayTotalOf?: boolean,
+  displayedName?: DisplayedName,
+  locale?: 'en' | 'fr',
+};
+
+const SearchContainer = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const TextContainer = styled(BasicText)`
+  display: flex;
+  flex-direction: column;
+  margin-right: ${spacing.sp8};
+`;
+
+const ResultContainer = styled(BasicText)`
+  font-weight: bold;
+`;
+
+const translations = {
+  en: {
+    search: 'Search',
+    total: 'Total',
+  },
+  fr: {
+    search: `Rechercher`,
+    total: `Total`,
+  },
+};
+
+export default function TableSearch(props: SearchProps) {
+  const {
+    onChange,
+    value = '',
+    displayTotalOf = true,
+    displayedName,
+    locale = 'en',
+    ...rest
+  } = props;
+  const { setGlobalFilter, rows } = useTableContext();
+
+  const totalDispayedRows = rows.length;
+
+  React.useEffect(() => {
+    setGlobalFilter(value);
+  }, [value, setGlobalFilter]);
+
+  return (
+    <SearchContainer>
+      {displayedName && displayTotalOf && (
```

Comment:
As `displayedName` and `displayTotalOf` are 2 dissociated props I believe that we should be able to use them in a separated way. So providing only `displayTotalOf` would display `Total: XX`, providing an additional `displayedName` would result in `Total: XX {displayedName}`.
If we take the opposite approach where we only want to display the total when the `displayedName` is provided then I think a single props should be exposed in the component definition : something like `displayTotalOf={{plural, singular}}` would be enough